test/basic.test.js

```diff
@@ -214,4 +214,16 @@ describe('basic support', () => {
   it('checks regex types', () => {
     expect(() => sift({$regex: {$mod2: 2}})).toThrow(TypeError);
   });
+
+  it('$exists properly checks non-object values', () => {
+    const falsePresence = sift({$exists: false});
```

Comment:
Shouldn't `void 0` in [this test](https://github.com/mixmaxhq/sift.js/blob/master/test/operations.test.js#L97) have covered this case? Why did that pass?

Reply:
@bradvogel no, in the current implementation `$exists: true/false` checks for the [key existence](https://github.com/mixmaxhq/sift.js/blob/master/src/utils.js#L59) in [passed object](https://github.com/mixmaxhq/sift.js/blob/master/src/index.js#L319), so in the test you mentioned it checks "if every array key exists", i.e.
```
{$exists: false}, [0, false, void 0, null], []
```
| expectedPresence      | value | key | object |
| ----------- | ----------- | ----------- | ----------- |
| `false`      | `0`       | `0`      | `[ 0, false, undefined, null ]` |
| `false`      | `false`       | `1`      | `[ 0, false, undefined, null ]` |
| `false`      | `undefined`       | `2`      | `[ 0, false, undefined, null ]` |
| `false`      | `null`       | `3`      | `[ 0, false, undefined, null ]` |

where value,key,object is basically `array.filter` call signature from the [below](https://github.com/mixmaxhq/sift.js/blob/master/test/operations.test.js#L332)

Reply:
But in case if you run
```
const truePresence = sift({$exists: true});
truePresence(42) // true
```
the signature is
```
expectedPresence = true
value = 42
key = undefined
object = undefined
```
Because there is no object to operate on

Reply:
@bradvogel another way of fixing this issue might be to work everywhere with object fields when using this filter, not with primitives, e.g. for the aforementioned PagerDuty alert we need to change [this line](https://github.com/mixmaxhq/rule-distribution/blob/11fc090a68a13085ace6754feed59c5b1bc671f0/src/utils/customRules/sift.js#L248) to be:
```
sift({
   numberOfOpens: params,
})({
   numberOfOpens: numOpens
})
```
(it can be any other name than `numberOfOpens`, but should be the same for both)

Reply:
I did run test with `$eq`, `$gt` and others using non-object values and they worked just fine, so it's probably a long-lasting bug from original `sift` package, because they even [had an example](https://github.com/crcn/sift.js/tree/v8.3.2#api) of using `$exists` on non-object values

I will merge this then and bump in `rule-distribution`, thanks for the review!